integr8ly/integreatly-operator
# INTLY-8421 Delete alerts during uninstall

# Description

Refactor the alert reconciliation logic for all the products, and add logic
to delete the alerts if the RHMI instanced is marked for deletion.

## Verification steps

In order to verify this PR, follow these steps:

1. Install RHMI on a cluster
2. Once installation is complete, verify that the *alerts have been created
3. Uninstall RHMI (delete the RHMI CR)
4. Verify that the *alerts are deleted

\* The following is a table with the alerts reconciled for each namespace:
| Product | Namespace | Alerts |
| ---------- | ---------------- | -------- |
| AMQ Online | `redhat-rhmi-amq-online` | `rhmi-amq-online-slo`, `ksm-endpoint-alerts`, `ksm-amqonline-alerts` |
| Apicurito | `redhat-rhmi-apicurito` | `ksm-apicurito-alerts`, `ksm-endpoint-alerts` |
| Apicurito | `redhat-rhmi-apicurito-operator` | `ksm-endpoint-alerts` |
| Cloud Resources Operator | `redhat-rhmi-cloud-resources-operator` | `ksm-endpoint-alerts` |
| CodeReady | `redhat-rhmi-codeready-workspaces` | `ksm-endpoint-alerts`, `ksm-codeready-alerts` |
| CodeReady | `redhat-rhmi-codeready-workspaces-operator` | `ksm-endpoint-alerts` |
| Fuse | `redhat-rhmi-fuse` | `ksm-endpoint-alert`, `ksm-fuse-online-alerts` |
| Fuse | `redhat-rhmi-fuse-operator` | `ksm-endpoint-alerts` |
| Monitoring | `redhat-rhmi-middleware-monitoring-operator` | `backup-monitoring-alerts`, `ksm-alerts`, `ksm-monitoring-alerts`, `ksm-endpoint-alerts` |
| SSO | `redhat-rhmi-rhsso` | `ksm-endpoint-alerts` |
| SSO | `redhat-rhmi-rhsso-operator` | `ksm-endpoint-alerts` |
| User SSO | `redhat-rhmi-user-sso` | `ksm-endpoint-alerts` |
| User SSO | `redhat-rhmi-user-sso-operator` | `ksm-endpoint-alerts` |
| Solution Explorer | `redhat-rhmi-solution-explorer` | `ksm-endpoint-alerts`, ``ksm-solution-explorer-alerts` |
| Solution Explorer | `redhat-rhmi-solution-explorer-operator` | `ksm-endpoint-alerts` |
| 3scale | `redhat-rhmi-3scale` | `kms-endpoint-alerts`, `ksm-3scale-alerts` |
| 3scale | `redhat-rhmi-3scale-operator` | `ksm-endpoint-alerts` |
| UPS | `redhat-rhmi-ups` | `ksm-endpoint-alerts` |
| UPS | `redhat-rhmi-ups-operator` | `ksm-endpoint-alerts` |

## Type of change


## Checklist
